### DISCUSSION TITLE
Comparisons to pymatgen StructureMatcher and comments on speed

### DISCUSSION THREAD

**Discussion by sgbaird** [ftherrien/p2ptrans#25] [Q&A] [answered]
Right now I'm using pymatgen's `StructureMatcher` and am looking for (hopefully) drop-in replacements to it due to speed concerns. See https://matsci.org/c/pymatgen/9 and https://github.com/sparks-baird/matbench-genmetrics/issues/9. It seems like `p2ptrans` might be one alternative to a drop-in replacement. Is my interpretation correct? Also, any comments on the speed? For example, running ~6e7 pairwise match calculations.

**Comment by ftherrien** [comment, marked as answer]
Hey @sgbaird , thanks for your question! I think the straightforward answers are:

1. Yes, you are correct p2ptrans can be used as a replacement for StructureMatcher
2. No you likely won't be able to run 6e7 match calculations in a reasonable amount of time.

p2ptrans is meant to find the shortest distance between two infinite lattice regardless of their representations (see why that can be problematic on Figure 1 of [our JCP paper](https://aip.scitation.org/doi/abs/10.1063/1.5131527) ). The focus is on the mapping more than on similarity. Although you *can* use it to compute a distance, if you are *only* interested in similarity I recommend you use a fingerprint based method as they do not necessitate the prohibitive mapping step, see [USPEX](https://pubmed.ncbi.nlm.nih.gov/20720316/)  for example. 

If you want to try p2ptrans, you can make it much faster if you don't care as much about finding the absolute shortest distance between two structures by (1) reducing the size of the mapping spheres (-n) and (2) reducing the number of random starts (n_iter). If you have relatively small structures you may get away with matching on the order of a second. This would be at your own risk as you would need to make sure that the results are consistent enough with what you would get with larger spheres and more random steps.

I hope this helps, let me know if you have more questions.

**Reply by sgbaird** [reply]
Thank you! I will keep `p2ptrans` in mind for future applications where the mapping is important, as opposed to simply a similarity metric. Very interesting work btw. Much appreciated!
